nanoFramework.WebServer/WebServer.cs

```diff
@@ -495,106 +495,81 @@ private void StartListener()
                 new Thread(() =>
                 {
                     bool isRoute = false;
-                    CallbackRoutes route;
-                    int urlParam;
-                    bool isFound;
-                    int incForSlash;
-                    string toCompare;
-                    string routeStr;
-                    string rawUrl;
+                    string rawUrl = context.Request.RawUrl;
+
+                    //This is for handling with transitory or bad requests
+                    if (rawUrl == null)
+                        return;
+
+                    int urlParam = rawUrl.IndexOf(ParamStart);
 
                     foreach (var rt in _callbackRoutes)
                     {
-                        route = (CallbackRoutes)rt;
-                        urlParam = context.Request.RawUrl.IndexOf(ParamStart);
-                        isFound = false;
-                        routeStr = route.Route;
-                        rawUrl = context.Request.RawUrl;
-                        incForSlash = routeStr.IndexOf('/') == 0 ? 0 : 1;
-                        toCompare = route.CaseSensitive ? rawUrl : rawUrl.ToLower();
-                        if (toCompare.IndexOf(routeStr) == incForSlash)
+                        CallbackRoutes route = (CallbackRoutes)rt;
+
+                        bool isFound;
+                        string routeStr = route.Route;
+                        int incForSlash = routeStr.IndexOf('/') == 0 ? 0 : 1;
+                        string toCompare = route.CaseSensitive ? rawUrl : rawUrl.ToLower();
+
+                        if (urlParam > 0)
+                            isFound = urlParam == routeStr.Length + incForSlash;
+                        else
+                            isFound = toCompare.Length == routeStr.Length + incForSlash;
+
+                        // todo - please help to name the variables: neededName, neededName2
```

Comment:
I'm fine with the neededName. For the seconf one neededMethod can be good. And maybe what's missing is just a comment?
something like:
// neededName = matching the route name
// neededMethod = matching the method type